### PR TITLE
Handle simultaneous touch events

### LINKED ISSUE #877: Bug - Safari iOS blocks all interactions but OpenseaDragon canvas
When using the library in Safari iOS 9.2 and after pinch zooming a bit, sometimes happens that the whole browser gets locked for user interactions. Though, the canvas where OpenseaDragon makes the drawing still works and reacts fine.
We first noticed this on a project we are developing, but then also tested it on your demo https://openseadragon.github.io/ and still got the same behaviour error.
Sometimes it happens that the browser comes back to life after a while and everything works again. But sometimes it doesn't, it gets blocked for good until the user refreshes the page.

The easiest way to reproduce seems to be to open it in the iOS simulator and stressing it by pinch zooming a lot. On the actual device (iPhones 5s, 6, 6s) it takes more tries to get the error to happen but it happens as well.

As always this is pretty urgent, so would really appreciate fast response :)
Thanks for an otherwise awesome library!

### PR DESCRIPTION
**Pull request by dlukez**
There is currently an issue with simultaneous touch handling, whereby the following steps cause the page to become unresponsive:

1. Place a single finger on the screen
2. Place another finger on the screen
3. Simultaneously remove both fingers at the same time (needs to be timed well)

I tested this on an iPhone 5s w/ iOS 9.2 (with some additional logging to make sure I was accurately releasing both fingers at the same time lol).

I'm not sure how this behaves on other devices... It'd be helpful if someone could verify this change doesn't break them.

Closes #877

### CODE REVIEW

**Comment by iangilman** [comment]
Wonderful! Looks like a good fix, but I'd like to ask @msalsbery to also take a look to verify  that he concurs. 

Thank you for diving into this!

**Comment by dlukez** [comment]
I'll look at adding unit tests for this over the weekend :)

**Comment by dlukez** [comment]
Added a script to reproduce using Chrome Device Emulation in https://github.com/openseadragon/openseadragon/issues/877#issuecomment-217370039

**Comment by iangilman** [comment]
Sorry for the delay... I've been out. Are you planning on adding tests, or did that turn out to not be possible (besides the Chrome script)?

**Comment by dlukez** [comment]
Yeah I'm planning on adding tests to this PR, just didn't get around to it last weekend. Looking into it now

**Comment by dlukez** [comment]
@iangilman I took my best shot at adding some unit tests given the way the other tests are set up. They should run properly in PhantomJS and I also tested in Chrome... I put some basic feature detection stuff for TouchEvents so hopefully the tests don't run and break in browsers without this functionality.

File: test/helpers/touch.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                false,                              // ctrlKey
+                false,                              // altKey
+                false,                              // shiftKey
+                false                               // metaKey
+            );
+            return touchEvent;
+        }
+    }
+
+})();
```
Comment (iangilman) [inline test/helpers/touch.js]: Git and GitHub like a blank line at the end of every file... please add one here.

File: test/test.html [hunk 2]
```diff
@@ -17,6 +17,7 @@
     <script src="/build/openseadragon/openseadragon.js"></script>
     <script src="/test/helpers/legacy.mouse.shim.js"></script>
     <script src="/test/helpers/test.js"></script>
+    <script src="/test/helpers/touch.js"></script>
```
Comment (iangilman) [inline test/test.html]: Please add this to coverage.html as well.

**Comment by iangilman** [comment]
@dlukez When you say they should run properly in PhantomJS, you mean they'll be skipped because Phantom doesn't have touch support? Or does it?

It would be great if our touch tests could run in our Travis builds.

At any rate, the test code looks good.

**Comment by dlukez** [comment]
Nope, I mean the test should actually run - PhantomJS has a touch API. You should be able to see the test if you run `grunt test --verbose` (look for *MouseTracker: touch events*).

**Comment by iangilman** [comment]
So it does! That's awesome! Thank you for adding these tests (and for fixing the issue in the first place)!

**Pull request merged**